src/Microsoft.Framework.PackageManager/Restore/NuGet/NuGetv3Feed.cs

```diff
@@ -0,0 +1,297 @@
+                        }
+
+                        try
+                        {
+                            JObject doc;
+                            using (var reader = new StreamReader(data.Stream))
+                            {
+                                doc = JObject.Load(new JsonTextReader(reader));
+                            }
+
+                            var result = doc["versions"]
+                                .Select(x => BuildModel(id, x.ToString()))
+                                .Where(x => x != null);
+
+                            results.AddRange(result);
+                        }
+                        catch
+                        {
+                            _reports.Information.WriteLine("The file {0} is corrupt",
+                                data.CacheFileName.Yellow().Bold());
+                            throw;
+                        }
+                    }
+
+                    return results;
+                }
+                catch (Exception ex)
+                {
+                    if (retry == 2)
+                    {
+                        // Fail silently by returning empty result list
+                        if (_ignoreFailure)
+                        {
+                            _ignored = true;
+                            _reports.Information.WriteLine(
+                                string.Format("Failed to retrieve information from remote source '{0}'",
+                                    _baseUri).Yellow().Bold());
+                            return new List<PackageInfo>();
+                        }
+
+                        _reports.Error.WriteLine(string.Format("Error: FindPackagesById: {1}\r\n  {0}",
+                            ex.Message, id).Red().Bold());
+                        throw;
+                    }
+                    else
+                    {
+                        _reports.Information.WriteLine(string.Format("Warning: FindPackagesById: {1}\r\n  {0}", ex.Message, id).Yellow().Bold());
+                    }
+                }
+            }
+            return null;
+        }
+
+        public PackageInfo BuildModel(string id, string version)
+        {
+            return new PackageInfo
+            {
+                // If 'Id' element exist, use its value as accurate package Id
+                // Otherwise, use the value of 'title' if it exist
+                // Use the given Id as final fallback if all elements above don't exist
+                Id = id,
+                Version = SemanticVersion.Parse(version),
+                ContentUri = _baseUri + id.ToLowerInvariant() + "/" + version.ToLowerInvariant() + "/" + id.ToLowerInvariant() + "." + version.ToLowerInvariant() + ".nupkg",
+            };
+        }
+
+        public async Task<Stream> OpenNuspecStreamAsync(PackageInfo package)
+        {
+            return await PackageUtilities.OpenNuspecStreamFromNupkgAsync(package, OpenNupkgStreamAsync, _reports.Information);
```

Comment:
That's not right for the V3 feed. The Nuspec is downloadable without getting the nupkg.